src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+                           high_frequency=channel[1], entity_type=protection_ent_type)
+
+    # Identify CBSD grants in the neighborhood of the protection point 
+    # and channel
+    neighborhood_grants = interf.findOverlappingGrantsInsideNeighborhood(
+                            grant_objects, channel_constraint)
+    num_grants = len(neighborhood_grants)
+
+    if num_grants is 0:
+      continue
+
+    # calculate the fair share
+    fair_share = interference_quota / num_grants
+
+    # Empty list of final IAP EIRP and interference assigned to the grant
+    grants_eirp = []
+    interference_list = []
+    # Initialize interference list to 0 for all the neighborhood_grants
+    interference_list[:num_grants] = [0] * num_grants
+
+    aggr_interference = 0
+
+    # Initialize flag to False for all the grants
+    grant_initialize_flag = []
+    grant_initialize_flag[:num_grants] = [False] * num_grants
+
+    # Initialize grant EIRP to maxEirp for all the grants 
+    grants_eirp = [grant.max_eirp for grant in neighborhood_grants]
+
+    while (num_grants > 0):
+      for index, grant in enumerate(neighborhood_grants):
+        if grant_initialize_flag[index] is False:
+          # Compute interference grant causes to protection point over channel
+
+          # Compute interference to FSS Co-channel protection constraint
+          if protection_ent_type is interf.ProtectedEntityType.FSS_CO_CHANNEL:
+            interference = interf.dbToLinear(interf.computeInterferenceFss(
+                             grant, channel_constraint, fss_info, grants_eirp[index]))
+
+          # Compute interference to FSS Blocking protection constraint
+          elif protection_ent_type is interf.ProtectedEntityType.FSS_BLOCKING:
+            interference = interf.dbToLinear(interf.computeInterferenceFssBlocking(
+                             grant, channel_constraint, fss_info, grants_eirp[index]))
+
+          # Compute interference to ESC protection constraint
+          elif protection_ent_type is interf.ProtectedEntityType.ESC_CAT_A or\
+                protection_ent_type is interf.ProtectedEntityType.ESC_CAT_B:
+            interference = interf.dbToLinear(interf.computeInterferenceEsc(
+                             grant, channel_constraint, esc_antenna_info, grants_eirp[index]))
+
+          # Compute interference to GWPZ or PPA protection constraint
+          else:
+            interference = interf.dbToLinear(interf.computeInterferencePpaGwpzPoint(
+                             grant, channel_constraint, interf.GWPZ_PPA_HEIGHT, 
+                             grants_eirp[index], region_type))
+
+          # calculated interference exceeds fair share of
+          # interference to which the grants are entitled
+          if interference < fair_share:
+            interference_list[index] = interference
+            # Grant is satisfied
+            grant_initialize_flag[index] = True
```

Comment:
Why not accumulating the satisfied grant interference here, since you pass here only once per satisfied grant (see my other comment below), and the aggregated interference amount ?
And updating num_grants? BTW this one should be renamed : num_unsatisfied_grants